src/work-allocation-2/components/case-manager-filter/case-manager-filter.component.ts

```diff
@@ -1,97 +1,152 @@
+      name: 'location',
+      options: locations.map(loc => ({key: loc.id, label: loc.locationName})),
+      minSelected: 1,
+      maxSelected: 1,
+      minSelectedError: 'You must select a location',
+      maxSelectedError: null,
+      title: 'Case Location',
+      type: 'select'
+    };
   }
 
-  private saveFilterToSession(): void {
-    const toStore = JSON.stringify({
-      locationId: this.location.id
-    });
-    this.sessionStorageService.setItem(FilterConstants.Session.TaskManager, toStore);
+  private static initRoleTypeFilter(): FilterFieldConfig {
+    return {
+      name: 'role',
+      options: [
+        {
+          key: PersonRole.JUDICIAL,
+          label: PersonRole.JUDICIAL
+        },
+        {
+          key: PersonRole.CASEWORKER,
+          label: PersonRole.CASEWORKER,
+        },
+        {
+          key: PersonRole.ADMIN,
+          label: PersonRole.ADMIN,
+        }
+      ],
+      minSelected: 1,
+      maxSelected: 1,
+      minSelectedError: 'You must select a role type',
+      maxSelectedError: null,
+      title: 'Role type',
+      type: 'select'
+    };
   }
 
-  public getLocationById(id: string): Location {
-    if (id === this.ALL_LOCATIONS.id) {
-      return this.ALL_LOCATIONS;
-    }
-    return this.locations.find(loc => loc.id === id) || this.ALL_LOCATIONS;
+  private static initPersonFilter(): FilterFieldConfig {
+    return {
+      name: 'selectPerson',
+      options: [
+        {
+          key: PersonRole.ALL,
+          label: PersonRole.ALL
+        },
+        {
+          key: 'Specific person',
+          label: 'Specific person'
+        }
+      ],
+      minSelected: 1,
+      maxSelected: 1,
+      minSelectedError: 'You must select a person',
+      maxSelectedError: null,
+      title: 'Person',
+      type: 'radio'
+    };
+  }
+
+  private static findPersonFilter(): FilterFieldConfig {
+    return {
+      name: 'person',
+      options: [],
+      minSelected: 0,
+      maxSelected: 0,
+      minSelectedError: 'You must select a person',
+      maxSelectedError: null,
+      showCondition: 'selectPerson=Specific person',
+      type: 'find-person'
+    };
   }
 
-  private emitChangedEvent(): void {
-    this.selectionChanged.emit({
-      location: this.location,
-      jurisdiction: this.jurisdiction,
+  public ngOnInit(): void {
+    this.fieldsConfig.fields = [
+      CaseManagerFilterComponent.initServiceFilter(),
+      CaseManagerFilterComponent.initCaseLocationFilter([{
+        id: 'all',
+        locationName: 'All locations',
+        services: []
+      }, ...this.locations]),
+      CaseManagerFilterComponent.initRoleTypeFilter(),
+      CaseManagerFilterComponent.initPersonFilter(),
+      CaseManagerFilterComponent.findPersonFilter()
+    ];
+    this.filterService.getStream(CaseManagerFilterComponent.FILTER_NAME)
+      .pipe(
+        filter((f: FilterSetting) => f && f.hasOwnProperty('fields'))
+      ).subscribe((f: FilterSetting) => {
     });
```

Comment:
if the suscribe block does nothing could that be removed? And also this whole filterService.getStream what is it used for as it's not assigned to any object.

Reply:
When you apply filter you get the filter settings object with the applied filters. This is what you send to the BE to filter down the list. I just logged them since the next ticket is sending over the params to the BE.